tests/test_numbers.py

```diff
@@ -1,66 +1,87 @@
 """Tests for all number generators."""
 
+import string
 import sys
+from collections import namedtuple
 
 import pytest
 
 from fauxfactory import (
+    gen_octagonal,
     gen_integer,
+    gen_hexadecimal,
     gen_negative_integer,
     gen_positive_integer,
 )
 
 
-def test_gen_integer_1():
-    """Create a random integer with no range limits."""
-    integer_types = (int,)
-    result = gen_integer()
-    assert isinstance(result, integer_types)
+GenFuncDataSet = namedtuple('GenFuncDataSet',
+                            ['gen_func', 'expect_type', 'base', 'valid_chars'])
+GenFuncDataSets = [
+    GenFuncDataSet(gen_octagonal, str, 8, string.octdigits + '-o'),
+    GenFuncDataSet(gen_integer, int, 10, string.digits + '-'),
+    GenFuncDataSet(gen_hexadecimal, str, 16, string.hexdigits + '-x')
+]
 
 
-def test_gen_integer_2():
-    """Create a random integer with set minimum limit."""
+@pytest.mark.parametrize('gen_func, expect_type, base, valid_chars',
```

Comment:
Since you're using a `namedtuple` perhaps you could only pass 1 argument and then access its attributes inside the code instead of naming all of them?